### PR TITLE
Improvement/reflux

### PR DESCRIPTION
**Pull request by ivarconr**
In the pull request I have rewritten the FeatureStore in to following the flux pattern with the [reflux](https://github.com/spoike/refluxjs) library.

So far I have rewritten featureToggles and archivedFeatureToggles. The main benefits are:
- Less logic in the view-components.
- Lower coupling
- "faster rendering" ass all data are available even when a component is not visible. 

If we move in this direction there is more work to be done:
- A separate store for the error-handling (aka ErrorStore). 
- ErrorHandling inside the store when polling data.
- StrategyStore must be converted to reflux.
- EventStore. Here we could also potentially start thinking about pagination and such. 

I hope people can look over the code and we can discuss whether we should move in this direction. 

(This pull request relates to issue #91. )

### CODE REVIEW

**Comment by Andersos** [comment]
Maybe @torarnek also can take a look.

File: public/js/components/feature/FeatureTogglesComponent.jsx [hunk 1]
```diff
@@ -1,38 +1,33 @@
...
     },
-
-    setFeatures: function (data) {
-        this.setState({features: data.features});
+    componentWillUnmount: function() {
+        this.unsubscribe();
     },
 
     handleError: function (error) {
+      console.log(error);
```
Comment (Andersos) [inline public/js/components/feature/FeatureTogglesComponent.jsx]: Do we want to console.log the error?

Comment (ivarconr) [inline public/js/components/feature/FeatureTogglesComponent.jsx]: Nope :)

**Comment by jarib** [comment]
Nice work!

**Comment by jarib** [comment]
I haven't used Reflux, but one thing that surprised me here is that server interaction is happening in `FeatureStore`. AFAIK, in a "traditional" Flux arhictecture a store would subscribe to events to update its state, not call out to the server itself. Server interaction would be handled in actions. Or perhaps I've misunderstood something?

**Comment by torarnek** [comment]
I have looked at FeatureToggleStore and its usage, and I think it looks superb! :)

@jarib You are right that the initial Flux architecture had interaction with server/api external to stores, however, with most variations of Flux, like Reflux, this is simplified, by just doing the server-interaction in the Stores.  It is of course possible to do it outside the stores, however, that would require an extra layer of actions, which might be unnecessary with these simple stores.

**Comment by ivarconr** [comment]
I could move all server-interactions/ajax-stuff inside the actions. 

I'm not sure how large the extra layer of actions would be, but I believe this would require us to have actions like "CreateFeatureToggleFailedAction" etc... Is this correct @torarnek ? 

Will this make the design simpler? Would it be easier to understand?

**Comment by ivarconr** [comment]
I was basically following this example: https://www.npmjs.com/package/reflux#asynchronous-actions-as-promises

**Comment by torarnek** [comment]
This is an interesting design question. Comparing doing the server-call in actions.js vs in store.js:

3 actions vs 1
Logic in two places(actions.js and store.js) vs logic in one place(store.js)

Are there any special benefits from doing the server-call in actions.js? Retry, undo etc? Anything that can justify 3x number of actions?

**Comment by ivarconr** [comment]
You make a valid point @torarnek. I think we should make the implementation as simple as possible for now. There is nothing stopping us moving in that direction later if we see any real benefits.

**Comment by msolli** [comment]
This has been discussed at length here: https://github.com/spoike/refluxjs/pull/140

I've had success keeping the server stuff in the actions, and triggering new actions (complete/error) when the result comes in. I find it makes for a nice flow of data and events, but at the cost of having more actions (ie. `searchLoad`, `searchLoadComplete`, `searchLoadError`). The store would listen to the `*Complete` actions, and an error store would listen to `*Error` actions.

Reflux has sugar for creating these "children" actions: https://github.com/spoike/refluxjs#asynchronous-actions

**Comment by ivarconr** [comment]
Thanks for your feedback @msolli. I think this is how we should implemented, mainly because it will keep the stores simpler.

**Comment by ivarconr** [comment]
Based on your great feedback I have implemented the stores for strategies, feature-toggles. archived-toggles and errors. 

Before merging this I will pull the initial-fetch and Timers out of the store into a separate module.

**Pull request merged**